### ISSUE TITLE
Allow using the current model as TASK_MODEL

### ISSUE THREAD

**Issue by lksj92hs** [huggingface/chat-ui#1556]
## Describe your feature request

Currently the task model (the model used for summarization, web search, etc.) is fixed with the `TASK_MODEL` environment variable and can't be changed without restarting chat-ui. It would be useful if we can use the current model as `TASK_MODEL` because:
* If we are using a large model for the current chat, it makes sense to use a larger task model as well to improve the quality of the results even with the cost of a bit more time waiting for a response. On the other hand, if we are using a smaller model, we won't have to pay the price of using a big model as task model.
*  For some chats the task model can be a privacy issue - e.g. if we are uploading sensitive documents, we may not want to use a public task model. Also there is a reasonable expectation by the user that his requests will be send only to the model we have explicitly selected with the current chat but they can be send to the task model as well without any indication or warning.

## Implementation idea

Implement a special value for the `TASK_MODEL` (e.g. `<current_model>` or something like this) to use the current chat model as task model instead of using a concrete task model.

**Comment by nsarrazin** [comment]
Hi, I think that would be a good idea. Maybe it even make sense that if TASK_MODEL is `''` or undefined we apply this behaviour? Currently it just takes the first model in the list which is pretty arbitrary.

**Comment by lksj92hs** [comment]
Yes, that would work fine too. The only issue is if this will break some backwards compatibility with the current behavior being that undefined or empty TASK_MODEL selects a specific default model.
